spec/requests/api/v1/customers/usage_spec.rb

```diff
@@ -0,0 +1,270 @@
+      end
+      let(:google_usa) do
+        create(:group, billable_metric: metric, key: 'region', value: 'usa', parent_group_id: google.id)
+      end
+
+      let(:charge) do
+        create(
+          :standard_charge,
+          plan: subscription.plan,
+          billable_metric: metric,
+          properties: {},
+          group_properties: [
+            build(
+              :group_property,
+              group: aws_usa,
+              values: { amount: '10', amount_currency: 'EUR' },
+            ),
+            build(
+              :group_property,
+              group: aws_france,
+              values: { amount: '20', amount_currency: 'EUR' },
+            ),
+            build(
+              :group_property,
+              group: google_usa,
+              values: { amount: '30', amount_currency: 'EUR' },
+            ),
+          ],
+        )
+      end
+
+      before do
+        create_list(
+          :event,
+          2,
+          organization:,
+          customer:,
+          subscription:,
+          code: metric.code,
+          timestamp: Time.zone.now,
+          properties: { cloud: 'aws', region: 'usa' },
+        )
+
+        create(
+          :event,
+          organization:,
+          customer:,
+          subscription:,
+          code: metric.code,
+          timestamp: Time.zone.now,
+          properties: { cloud: 'aws', region: 'france' },
+        )
+
+        create(
+          :event,
+          organization:,
+          customer:,
+          subscription:,
+          code: metric.code,
+          timestamp: Time.zone.now,
+          properties: { cloud: 'google', region: 'usa' },
+        )
+      end
+
+      it 'returns the group usage for the customer' do
+        get_with_token(
+          organization,
+          "/api/v1/customers/#{customer.external_id}/current_usage?external_subscription_id=#{subscription.external_id}",
+        )
+
+        charge_usage = json[:customer_usage][:charges_usage].first
+        groups_usage = charge_usage[:groups]
+
+        aggregate_failures do
+          expect(charge_usage[:units]).to eq('4.0')
+          expect(charge_usage[:amount_cents]).to eq(7000)
+          expect(groups_usage).to contain_exactly(
+            {
+              lago_id: aws_usa.id,
+              key: 'aws',
+              value: 'usa',
+              units: '2.0',
+              amount_cents: 2000,
+            },
+            { lago_id: aws_france.id, key: 'aws', value: 'france', units: '1.0', amount_cents: 2000 },
+            { lago_id: google_usa.id, key: 'google', value: 'usa', units: '1.0', amount_cents: 3000 },
+          )
+        end
+      end
+    end
+
+    context 'when customer does not belongs to the organization' do
+      let(:customer) { create(:customer) }
+
+      it 'returns not found' do
+        get_with_token(
+          organization,
+          "/api/v1/customers/#{customer.external_id}/current_usage?external_subscription_id=#{subscription.external_id}",
```

Comment:
Layout/LineLength: Line is too long. [121/120] (https://rubystyle.guide#max-line-length)